### PR TITLE
Minimal bufferWithCount demos cleanup

### PR DESCRIPTION
**Pull request by asm89**
Instead of having bigger demo files that demo multiple modes of
operation. Let's keep the demos short. For the expected output folks
should look at the `.expect` file. Also make the stdout observer a bit
smarter by making it json encode all array values before outputting
them.

### CODE REVIEW

File: demo/bootstrap.php [hunk 1]
```diff
@@ -15,9 +15,16 @@
     throw new RuntimeException('Install dependencies to run test suite.');
 }
 
+function asString($value) {
+    if (is_array($value)) {
+        return json_encode($value);
```
Comment (Ocramius) [inline demo/bootstrap.php:20]: seems kinda arbitrary? :|

File: demo/bootstrap.php [hunk 2]
```diff
@@ -15,9 +15,16 @@
...
+    }
+    return (string) $value;
```
Comment (Ocramius) [inline demo/bootstrap.php:22]: Should probably check with other types of value as well. What about objects?

Comment (Ocramius) [inline demo/bootstrap.php:22]: EDIT: nvm, saw that these are the demo files

File: demo/bootstrap.php (hunk 1 above)
Comment (asm89) [inline demo/bootstrap.php:20]: It is. :+1:

**Pull request merged**